app/src/main/java/com/arctouch/codechallenge/ui/details/MovieDetailsViewModel.kt

```diff
@@ -0,0 +1,87 @@
+package com.arctouch.codechallenge.ui.details
+
+import android.arch.lifecycle.MutableLiveData
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.api.TmdbApi
+import com.arctouch.codechallenge.base.BaseViewModel
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.util.API_KEY
+import com.arctouch.codechallenge.util.DEFAULT_LANGUAGE
+import com.arctouch.codechallenge.util.MovieImageUrlBuilder
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import javax.inject.Inject
+
+class MovieDetailsViewModel : BaseViewModel() {
+    @Inject
+    lateinit var tmdbApi: TmdbApi
+
+    private val movieTitle = MutableLiveData<String>()
+    private val movieGenres = MutableLiveData<String>()
+    private val movieReleaseDate = MutableLiveData<String>()
+    private val moviePosterPath = MutableLiveData<String>()
+    private val movieBackdropPath = MutableLiveData<String>()
+    private val movieOverview = MutableLiveData<String>()
+    private val movieImageUrlBuilder = MovieImageUrlBuilder()
+
+    private lateinit var subscription: Disposable
+
+    val errorMessage: MutableLiveData<Int> = MutableLiveData()
+
+    fun get(movieId: Long) {
+        subscription = tmdbApi.movie(movieId, API_KEY, DEFAULT_LANGUAGE)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doOnSubscribe { onRetrievePartnerListStart() }
+                .doOnTerminate { onRetrievePartnerListFinish() }
+                .subscribe(
+                        { result -> bind(result) },
+                        { onRetrievePartnerListError() }
+                )
+    }
+
+    private fun bind(movie: Movie) {
+        movieTitle.value = movie.title
+        movieGenres.value = movie.genres?.joinToString(separator = ", ") { it.name }
+        movieReleaseDate.value = movie.releaseDate
+        moviePosterPath.value = movie.posterPath?.let { movieImageUrlBuilder.buildPosterUrl(it) }
+        movieBackdropPath.value = movie.backdropPath?.let { movieImageUrlBuilder.buildBackdropUrl(it) }
+        movieOverview.value = movie.overview
+    }
+
+    fun getMovieTitle(): MutableLiveData<String> {
+        return movieTitle
+    }
+
+    fun getMovieGenres(): MutableLiveData<String> {
+        return movieGenres
+    }
+
+    fun getMovieReleaseDate(): MutableLiveData<String> {
+        return movieReleaseDate
+    }
+
+    fun getMoviePosterPath(): MutableLiveData<String> {
+        return moviePosterPath
+    }
+
+    fun getMovieBackdropPath(): MutableLiveData<String> {
+        return movieBackdropPath
+    }
+
+    fun getMovieOverview(): MutableLiveData<String> {
+        return movieOverview
+    }
+
+    private fun onRetrievePartnerListStart() {
+        errorMessage.value = null
+    }
+
+    internal fun onRetrievePartnerListFinish() {
```

Comment:
AndroidStudio tells me this method could be `private`. Why did you make it `internal`? Do you think the code would improve by making that change?

And why is the method here if it's empty? What do you think of leaving unused code in the project?

Reply:
It was a misspelling mistake. I have made it internal for another situation that I was testing but forgot to remove it. Leaving unused code is bad in many ways, it makes the code readability bad and confuse.